frontend/src/lib/components/Sharing/SharingModal.tsx

```diff
@@ -174,9 +174,10 @@ export function SharingModalContent({
                                                         </div>
                                                     }
                                                     onChange={() =>
-                                                        guardAvailableFeature(AvailableFeature.WHITE_LABELLING, () =>
+                                                        guardAvailableFeature(AvailableFeature.WHITE_LABELLING, () => {
                                                             onChange(!value)
-                                                        )
+                                                            setIsWhitelabelEnabled(!value)
```

Comment:
There's probably a better way to do this instead of calling the action here. 

I thought about adding onChange to the listeners but then it will always be triggered on any change in the form and i don't want that.